clusterloader2/testing/load/config.yaml

```diff
@@ -477,283 +320,98 @@ steps:
-    objectBundle:
-      - basename: medium-job
-        objectTemplatePath: job.yaml
-        templateFillMap:
-          ReplicasMin: {{MultiplyInt $MEDIUM_GROUP_SIZE 0.5}}
-          ReplicasMax: {{MultiplyInt $MEDIUM_GROUP_SIZE 1.5}}
-  - namespaceRange:
-      min: 1
-      max: {{$namespaces}}
-    replicasPerNamespace: {{$bigJobsPerNamespace}}
-    tuningSet: RandomizedScalingTimeLimited
-    objectBundle:
-      - basename: big-job
-        objectTemplatePath: job.yaml
-        templateFillMap:
-          ReplicasMin: {{MultiplyInt $BIG_GROUP_SIZE 0.5}}
-          ReplicasMax: {{MultiplyInt $BIG_GROUP_SIZE 1.5}}
+- module:
+    path: /modules/reconcile-objects.yaml
+    params:
+      actionName: "scaling and updating"
+      namespaces: {{$namespaces}}
+      randomScaleFactor: 0.5
+      daemonSetImage: k8s.gcr.io/pause:3.1
+      daemonSetReplicas: 1
+      bigDeploymentSize: {{$BIG_GROUP_SIZE}}
+      bigDeploymentsPerNamespace: {{$bigDeploymentsPerNamespace}}
+      mediumDeploymentSize: {{$MEDIUM_GROUP_SIZE}}
+      mediumDeploymentsPerNamespace: {{$mediumDeploymentsPerNamespace}}
+      smallDeploymentSize: {{$SMALL_GROUP_SIZE}}
+      smallDeploymentsPerNamespace: {{$smallDeploymentsPerNamespace}}
+      smallStatefulSetSize: {{$SMALL_GROUP_SIZE}}
+      smallStatefulSetsPerNamespace: {{$SMALL_STATEFUL_SETS_PER_NAMESPACE}}
+      mediumStatefulSetSize: {{$MEDIUM_GROUP_SIZE}}
+      mediumStatefulSetsPerNamespace: {{$MEDIUM_STATEFUL_SETS_PER_NAMESPACE}}
+      bigJobSize: {{$BIG_GROUP_SIZE}}
+      bigJobsPerNamespace: 1
+      mediumJobSize: {{$MEDIUM_GROUP_SIZE}}
+      mediumJobsPerNamespace: 1
+      smallJobSize: {{$SMALL_GROUP_SIZE}}
+      smallJobsPerNamespace: 1
 
-- name: Waiting for objects to become scaled
-  measurements:
-  - Method: WaitForControlledPodsRunning
-    Instances:
-    - Identifier: WaitForRunningDeployments
-    - Identifier: WaitForRunningStatefulSets
-    - Identifier: WaitForRunningDaemonSets
-    - Identifier: WaitForRunningJobs
-    Params:
-      action: gather
+- module:
+    path: /modules/reconcile-objects.yaml
+    params:
+      actionName: "deleting"
+      namespaces: {{$namespaces}}
+      daemonSetImage: k8s.gcr.io/pause:3.1
+      daemonSetReplicas: 0
+      bigDeploymentSize: {{$BIG_GROUP_SIZE}}
+      bigDeploymentsPerNamespace: 0
+      mediumDeploymentSize: {{$MEDIUM_GROUP_SIZE}}
+      mediumDeploymentsPerNamespace: 0
+      smallDeploymentSize: {{$SMALL_GROUP_SIZE}}
+      smallDeploymentsPerNamespace: 0
+      smallStatefulSetSize: {{$SMALL_GROUP_SIZE}}
+      smallStatefulSetsPerNamespace: 0
+      mediumStatefulSetSize: {{$MEDIUM_GROUP_SIZE}}
+      mediumStatefulSetsPerNamespace: 0
+      bigJobSize: {{$BIG_GROUP_SIZE}}
+      bigJobsPerNamespace: 0
+      mediumJobSize: {{$MEDIUM_GROUP_SIZE}}
+      mediumJobsPerNamespace: 0
+      smallJobSize: {{$SMALL_GROUP_SIZE}}
+      smallJobsPerNamespace: 0
 
-- name: Deleting objects
+{{if $ENABLE_PVS}}
+- name: Deleting PVCs
```

Comment:
Hm, do I see correctly that before deleting PVCs was together with deleting other resources? 
If that's the case then splitting it into separate step will actually make our test longer and also less demanding.

Reply:
Fixed. The diff is now simpler: https://gist.github.com/jupblb/7517fe2913b7369346494cc9a605ebe7